src/geo/map/cartodb-layer.js

```diff
@@ -25,6 +25,11 @@ var CartoDBLayer = LayerModelBase.extend({
       this.set('initialStyle', attrs.cartocss);
     }
 
+    // TODO: Validate source?
+    if (attrs.source) {
```

Comment:
are we going to admit this? I mean, I thought that sources can only be set with `setSource`

Reply:
They should be mandatory when creating the layer and they can only be replaced via `setSource`. We have [this check to ensure dataviews get a `source` when they are initialised](https://github.com/CartoDB/cartodb.js/blob/v4/src/dataviews/dataview-model-base.js#L92). I think we should do the same here. Agree?

Reply:
yep, we must check source for cartolayers and torquelayers